dask/bag/core.py

```diff
@@ -1134,7 +1134,7 @@ def partition(grouper, sequence, npartitions, p, nelements=2**20):
         d2 = defaultdict(list)
         for k, v in d.items():
             d2[abs(hash(k)) % npartitions].extend(v)
-        p.append(d2)
+        p.append(d2, fsync=True)
```

Comment:
I'll post our reproducible example next time I'm near a laptop. Short version is: if you apply a func to a group by using on-disk shuffle with multiple workers, the resultant data frame is missing rows. The number of rows it has is not reproducible. The problem does not occur with multiprocessing or of course with task-based shuffle.